File: app/src/main/java/com/sdex/activityrunner/db/application/ApplicationModel.java
package com.sdex.activityrunner.db.application;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.support.annotation.NonNull;

import java.io.Serializable;

@Entity(primaryKeys = {"packageName"})
public class ApplicationModel implements Serializable {

  public static final String TABLE = "ApplicationModel";
  public static final String NAME = "name";
  public static final String PACKAGE_NAME = "packageName";
  public static final String ACTIVITIES_COUNT = "activitiesCount";
  public static final String EXPORTED_ACTIVITIES_COUNT = "exportedActivitiesCount";

  @NonNull
  private String name;
  @NonNull
  private String packageName;
  @Deprecated
  @NonNull
  private String iconPath;
  @NonNull
  private int activitiesCount;
  @NonNull
  private int exportedActivitiesCount;

  @Deprecated
  public ApplicationModel(@NonNull String name, @NonNull String packageName,
                          @NonNull String iconPath) {
    this.name = name;
    this.packageName = packageName;
    this.iconPath = iconPath;
  }

  @Ignore
  public ApplicationModel(@NonNull String name, @NonNull String packageName) {
    this(name, packageName, "");
  }

  @NonNull
  public String getName() {
    return name;
  }

  @NonNull
  public String getPackageName() {
    return packageName;
  }

  @Deprecated
  @NonNull
  public String getIconPath() {
    return iconPath;
  }

  @NonNull
  public int getActivitiesCount() {
    return activitiesCount;
  }

  public void setActivitiesCount(@NonNull int activitiesCount) {
    this.activitiesCount = activitiesCount;
  }

  @NonNull
  public int getExportedActivitiesCount() {
    return exportedActivitiesCount;
  }

  public void setExportedActivitiesCount(@NonNull int exportedActivitiesCount) {
    this.exportedActivitiesCount = exportedActivitiesCount;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    ApplicationModel that = (ApplicationModel) o;
    return packageName.equals(that.packageName);
  }

  @Override
  public int hashCode() {
    return packageName.hashCode();
  }
}

File: app/src/main/java/com/sdex/activityrunner/db/application/ApplicationModel.kt
package com.sdex.activityrunner.db.application

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore

import java.io.Serializable

@Entity(primaryKeys = ["packageName"])
class ApplicationModel @Deprecated("")
constructor(val name: String, val packageName: String,
            @Deprecated("")
            @get:Deprecated("")
            val iconPath: String) : Serializable {
  var activitiesCount: Int = 0
  var exportedActivitiesCount: Int = 0

  @Ignore
  constructor(name: String, packageName: String) : this(name, packageName, "")

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other == null || javaClass != other.javaClass) return false
    val that = other as ApplicationModel?
    return packageName == that!!.packageName
  }

  override fun hashCode(): Int {
    return packageName.hashCode()
  }

  companion object {

    const val TABLE = "ApplicationModel"
    const val NAME = "name"
    const val PACKAGE_NAME = "packageName"
    const val ACTIVITIES_COUNT = "activitiesCount"
    const val EXPORTED_ACTIVITIES_COUNT = "exportedActivitiesCount"
  }
}
